test-requirements.txt

```
@@ -1,3 +1,11 @@
+# https://github.com/OCA/margin-analysis/pull/76
+git+https://github.com/legalsylvain/margin-analysis@12.0-MIG-product_margin_classification#subdirectory=setup/product_margin_classification
+
+# https://github.com/OCA/purchase-workflow/pull/1185
+git+https://github.com/legalsylvain/purchase-workflow@12.0-purchase_xxx_discount-apply-default-value#subdirectory=setup/purchase_discount
+git+https://github.com/legalsylvain/purchase-workflow@12.0-purchase_xxx_discount-apply-default-value#subdirectory=setup/purchase_triple_discount
```

Note this works too and helps removing a comment. For PRs you control this might be useful. For PR you don't control, the drawback is less stability in case the author rebases or something.

```suggestion
git+https://github.com/OCA/purchase-workflow@refs/pull/1185/head#subdirectory=setup/purchase_triple_discount
```

---

thanks a lot for taking time to take a look on my CI problems ! 

For PR I don't control (quite theoritical case, it doesn't occure in fact), if I understand correctly git / github, comparing : 

A) ``git+https://github.com/acsone/purchase-workflow@12.0-BRANCH-NAME#subdirectory=setup/MODULE``

B) ``git+https://github.com/OCA/purchase-workflow@refs/pull/xxx/head#subdirectory=setup/MODULE``

if acsone delete the branch ``12.0-BRANCH-NAME``, B should still work (event if the PR is closed) but A will fail, right ?



Second question, I have a lot of dependency in that repo, of my incubator repo. It seems it take a while to fetch all the code on the github CI with such requirements : 
```
git+https://github.com/grap/grap-odoo-incubator@12.0#subdirectory=setup/name_search_reset_res_partner
git+https://github.com/grap/grap-odoo-incubator@12.0#subdirectory=setup/purchase_package_qty
git+https://github.com/grap/grap-odoo-incubator@12.0#subdirectory=setup/mobile_kiosk_abstract
```

do you know if there are a syntax to do something like that, to fetch many modules ?

```
git+https://github.com/grap/grap-odoo-incubator@12.0#subdirectory=setup/name_search_reset_res_partner,setup/purchase_package_qty,setup/mobile_kiosk_abstract
```

---

> if acsone delete the branch 12.0-BRANCH-NAME, B should still work (event if the PR is closed) but A will fail, right ?

Yes that is correct. Although with B you are still at risk of acsone changing the PR with commits you don't want.

> Second question, I have a lot of dependency in that repo, of my incubator repo. It seems it take a while to fetch all the code on the github CI with such requirements :

Yes I'm aware of that. I had an optimization that broke with v15 and that I need to re-do.